### PR TITLE
WIP: Add audit sorting

### PR DESCRIPTION
**Pull request by cbaines**
Trello: https://trello.com/c/gs4d7GDP/267-3-add-additional-sort-criteria-on-content-list-page

### CODE REVIEW

**Comment by tuzz** [comment]
This is looking good so far!

File: app/domain/search/sort.rb [hunk 1]
```diff
@@ -4,31 +4,56 @@ class Sort
...
+          # value for the next item could be null, or it might not be.
+          scope
+        when :last
+          scope.where("#{field} IS NULL")
+        end
+      else
+        scope.where(
+          "#{field} #{{ desc: '<=', asc: '>=' }[order]} ?", field_value
+        )
+      end
```
Comment (tuzz) [inline app/domain/search/sort.rb]: What does this code do? What's the issue with NULL values?

**Comment by cbaines** [comment]
I'll leave this as WIP until the direction is clearer on what happens with the "next" functionality. This branch currently includes my first implementation of the improvements to that.

File: app/domain/search/sort.rb (hunk 1 above)
Comment (cbaines) [inline app/domain/search/sort.rb]: The `where_only_after` function takes a scope and content_item, and returns a new scope based on the one provided, but where the scope has been narrowed as much as possible to only include items after the provided item.

From the Sort class, you know the field that is being used for the sort, and the order. From the provided content item, you know the value of the field which is being used in the sort. From these three bits of information, you can generate a new condition to apply to the scope to further limit it to only items which have a sort value equal to or after (this doesn't quite match the naming, it needs weakening).

This approach won't work where the value of the field that the sort is using is nil, as that will be `NULL` in the query. However, the sort includes the information on where the null values are placed. If they are first, then I don't think there is a way of narrowing down the possibilities, so just return the scope. If the null values are last, you can at least eliminate anything that is not null, so do that.

Does that make some amount of sense?

**Comment by pmanrubia** [comment]
I am closing this PR. We will continue with this work once this feature has been prioritised again.

**Pull request closed without merging**